drivers/dummy/rasterizer_dummy.h

```diff
@@ -0,0 +1,622 @@
+
+	void gi_probe_set_to_cell_xform(RID p_probe, const Transform &p_xform) {}
+	Transform gi_probe_get_to_cell_xform(RID p_probe) const { return Transform(); }
+
+	void gi_probe_set_dynamic_data(RID p_probe, const PoolVector<int> &p_data) {}
+	PoolVector<int> gi_probe_get_dynamic_data(RID p_probe) const {
+		PoolVector<int> p;
+		return p;
+	}
+
+	void gi_probe_set_dynamic_range(RID p_probe, int p_range) {}
+	int gi_probe_get_dynamic_range(RID p_probe) const { return 0; }
+
+	void gi_probe_set_energy(RID p_probe, float p_range) {}
+	float gi_probe_get_energy(RID p_probe) const { return 0.0; }
+
+	void gi_probe_set_bias(RID p_probe, float p_range) {}
+	float gi_probe_get_bias(RID p_probe) const { return 0.0; }
+
+	void gi_probe_set_normal_bias(RID p_probe, float p_range) {}
+	float gi_probe_get_normal_bias(RID p_probe) const { return 0.0; }
+
+	void gi_probe_set_propagation(RID p_probe, float p_range) {}
+	float gi_probe_get_propagation(RID p_probe) const { return 0.0; }
+
+	void gi_probe_set_interior(RID p_probe, bool p_enable) {}
+	bool gi_probe_is_interior(RID p_probe) const { return false; }
+
+	void gi_probe_set_compress(RID p_probe, bool p_enable) {}
+	bool gi_probe_is_compressed(RID p_probe) const { return false; }
+
+	uint32_t gi_probe_get_version(RID p_probe) { return 0; }
+
+	GIProbeCompression gi_probe_get_dynamic_data_get_preferred_compression() const { return GI_PROBE_UNCOMPRESSED; }
+	RID gi_probe_dynamic_data_create(int p_width, int p_height, int p_depth, GIProbeCompression p_compression) { return RID(); }
+	void gi_probe_dynamic_data_update(RID p_gi_probe_data, int p_depth_slice, int p_slice_count, int p_mipmap, const void *p_data) {}
+
+	/* LIGHTMAP CAPTURE */
+	struct LightmapCaptureOctree {
+
+		enum {
+			CHILD_EMPTY = 0xFFFFFFFF
+		};
+
+		uint16_t light[6][3]; //anisotropic light
+		float alpha;
+		uint32_t children[8];
+	};
+
+	RID lightmap_capture_create() { return RID(); }
+	void lightmap_capture_set_bounds(RID p_capture, const AABB &p_bounds) {}
+	AABB lightmap_capture_get_bounds(RID p_capture) const { return AABB(); }
+	void lightmap_capture_set_octree(RID p_capture, const PoolVector<uint8_t> &p_octree) {}
+	PoolVector<uint8_t> lightmap_capture_get_octree(RID p_capture) const {
+		PoolVector<uint8_t> p;
+		return p;
+	}
+	void lightmap_capture_set_octree_cell_transform(RID p_capture, const Transform &p_xform) {}
+	Transform lightmap_capture_get_octree_cell_transform(RID p_capture) const { return Transform(); }
+	void lightmap_capture_set_octree_cell_subdiv(RID p_capture, int p_subdiv) {}
+	int lightmap_capture_get_octree_cell_subdiv(RID p_capture) const { return 0; }
+	void lightmap_capture_set_energy(RID p_capture, float p_energy) {}
+	float lightmap_capture_get_energy(RID p_capture) const { return 0.0; }
+	const PoolVector<RasterizerStorage::LightmapCaptureOctree> *lightmap_capture_get_octree_ptr(RID p_capture) const {
+		PoolVector<RasterizerStorage::LightmapCaptureOctree> p;
+		return &p;
```

Comment:
> > Is there memory leak / memory allocation bug here?

No, but it returns an invalid pointer that might crash the engine or corrupts its memory if used

Reply:
How would you fix it?

Reply:
Not sure, probably the safest way is to implement lightmap storage properly, like I did for texture (i.e. implement create, owner, and the like)